components/Articles.js

```diff
@@ -3,13 +3,13 @@ import { useChannel } from "@ably-labs/react-hooks";
 import ArticlePreview from "./ArticlePreview";
 import styles from "../styles/Home.module.css";
 
-export default function Articles() {
+export default function Articles(props) {
   let inputBox = null;
 
   const [headlineText, setHeadlineText] = useState("");
-  const [headlines, updateHeadlines] = useState([]);
-  const [_, ably] = useChannel("[?rewind=3]headlines", (headline) => {
-    updateHeadlines((prev) => [...prev, headline]);
+  const [headlines, updateHeadlines] = useState(props.history);
+  const [_, ably] = useChannel("headlines", (headline) => {
```

Comment:
See top-level comments about ordering etc. If we have concerns about this I'd also keep the history subscription (of 5 to be consistent with the SSR?) here to ensure we don't miss messages during the load time.

Reply:
Done in [a65cd39](https://github.com/ably-labs/ably-next-vercel-news/pull/8/commits/a65cd397ed945fe13c86d64b8f36fffaa563f98b)